katas/content/superposition/parity_bitstrings/solution.md

```diff
@@ -0,0 +1,56 @@
+There are multiple ways to approach this problem. In our first solution, we are going to use a recursive approach that was hinted at in the task.
+
+Let's denote the required state on $N$ qubits as $|P_{N,0}\rangle$ for `parity = 0` and $|P_{N,1}\rangle$ for `parity = 1`. We can group the $2^{N-1}$ basis states included in the state $|P_{N,0}\rangle$ by their first bit ($0$ or $1$) and write the state as follows:
+
+$$|P_{N,0}\rangle = \frac{1}{\sqrt{2^{N-1}}} \sum_{k : k \text{ has parity }0} |k\rangle_N = $$
+$$= \frac{1}{\sqrt{2^{N-1}}} \big( |0\rangle \otimes \sum_{k' : k' \\text{ has parity }0} |k'\rangle_{N-1} + |1\rangle \otimes \sum_{k'' : k'' \\text{ has parity }1} |k''\rangle_{N-1} \big) = $$
```

Comment:
These won't affect the current parser but the change in an adjacent pull request starts treating these as new lines, so let's preempt that and use single-backslashes:
```suggestion
$$= \frac{1}{\sqrt{2^{N-1}}} \big( |0\rangle \otimes \sum_{k' : k' \text{ has parity }0} |k'\rangle_{N-1} + |1\rangle \otimes \sum_{k'' : k'' \text{ has parity }1} |k''\rangle_{N-1} \big) = $$
```